lib/Window.js

```diff
@@ -23,105 +23,205 @@ module.exports = (Item) => {
+          break
+
+        case 1:
+          assert.ok(mouseButton <= 1, 'invalid operation')
+          this.shiftClick(click)
+          break
+
+        case 2:
+          assert.ok(mouseButton <= 8, 'invalid operation')
+          this.numberClick(click)
+          break
+
+        case 3:
+          assert.ok(mouseButton === 2, 'invalid operation')
+          this.middleClick(click)
+          break
+
+        case 4:
+          assert.ok(mouseButton <= 1, 'invalid operation')
+          this.dropClick(click)
+          break
+
+        case 5:
+          assert.ok([1, 5, 9, 2, 6, 10].includes(mouseButton), 'invalid operation')
+          this.dragClick(click)
+          break
+
+        case 6:
+          assert.ok(mouseButton === 0, 'invalid operation')
+          this.doubleClick(click)
+          break
+      }
+    }
+
+    mouseClick (click) {
       if (click.slot === -999) {
-        return this.acceptOutsideWindowClick(click)
-      } else if (click.slot >= this.inventoryStart && click.slot < this.inventoryEnd) {
-        return this.acceptInventoryClick(click)
-      } else if (click.slot === this.craftingResultSlot) {
-        return this.acceptCraftingClick(click)
+        this.dropSelectedItem(click.mouseButton === 0)
       } else {
-        return this.acceptNonInventorySwapAreaClick(click)
+        let { item } = click
+        if (click.mouseButton === 0) { // left click
+          if (item && this.selectedItem) {
+            if (Item.equal(item, this.selectedItem, false)) {
+              if (click.slot === this.craftingResultSlot) {
+                if (item.count + this.selectedItem.count > item.stackSize) {
+                  this.selectedItem.count += item.count
+                  this.updateSlot(item.slot, null)
+                }
+              } else {
+                this.fillSlotWithSelectedItem(item, true)
+              }
+            } else {
+              this.swapSelectedItem(click.slot, item)
+            }
+          } else if (this.selectedItem || item) {
+            this.swapSelectedItem(click.slot, item)
+          }
+        } else { // right click
+          if (this.selectedItem) {
+            if (item) {
+              if (Item.equal(item, this.selectedItem, false)) {
+                this.fillSlotWithSelectedItem(item, false)
+              } else {
+                this.swapSelectedItem(click.slot, item)
+              }
+            } else {
+              item = new Item(this.selectedItem.type, 0, this.selectedItem.metadata, this.selectedItem.nbt)
+              this.updateSlot(click.slot, item)
+              this.fillSlotWithSelectedItem(item, false)
+            }
+          } else if (item) {
+            if (click.slot !== this.craftingResultSlot) {
+              this.splitSlot(item)
+            } else {
+              this.swapSelectedItem(click.slot, item)
+            }
+          }
+        }
       }
     }
 
-    acceptOutsideWindowClick (click) {
```

Comment:
Changing all the already implemented function to new functions with different names is a breaking change. We won't be able to release this without a major version jump or this will break everything that is using prismarine-windows.

Reply:
As stated above "they were hard to read and inflexible"

The functions that were removed should only be called by acceptClick() and are probably also not called from anywhere outside, as the removed functions simply split up the logic of acceptClick()